File: binance4j-portfolio-margin/src/test/java/com/binance4j/portfoliomargin/client/PortfolioMarginClientTest.java
package com.binance4j.portfoliomargin.client;

import org.junit.jupiter.api.Test;

import com.binance4j.core.exception.ApiException;
import com.binance4j.core.test.CustomTest;

public class PortfolioMarginClientTest extends CustomTest {
	protected PortfolioMarginClient client = new PortfolioMarginClient(key, secret);

	@Override
	protected PortfolioMarginClient getClient() {
		return client;
	}

	// @Test Unauthorized?!
	void testGetAccountInfo() throws ApiException {
		testNotThrow(getClient().getAccountInfo());
	}

	@Test
	void testGetCollateralRate() throws ApiException {
		testNotThrow(getClient().getCollateralRate());
	}
}

File: src/test/com/binance4j/connectors/portfoliomargin/client/PortfolioMarginClientTest.kt
package com.binance4j.connectors.portfoliomargin.client

import com.binance4j.connectors.Connectors
import com.binance4j.connectors.core.test.CustomTest
import org.junit.jupiter.api.Test

internal class PortfolioMarginClientTest : CustomTest() {
    private var client = Connectors.rest(key, secret).portfolio

    //@Test TODO Don't have account ...
    //fun getAccountInfo() = assertDoesNotThrow(client.getAccountInfo())

    @Test
    fun getCollateralRate() = assertDoesNotThrow(client.getCollateralRate())

    @Test
    fun getBankruptcyLoanAmount() = assertDoesNotThrow(client.getBankruptcyLoanAmount())

    //@Test TODO can't repay, no account.
    //fun repay() = assertDoesNotThrow(client.repay())
}